task/control-plane-agent/src/mgs_sidecar.rs

```diff
@@ -166,19 +237,204 @@ impl MgsHandler {
 
         Ok(state)
     }
+
+    /// Unlocks the tech port if the challenge and response are compatible
+    fn unlock(
+        &mut self,
+        vid: u16,
+        challenge: UnlockChallenge,
+        response: UnlockResponse,
+        time_sec: u32,
+    ) -> Result<(), GwMonorailError> {
+        ringbuf_entry!(Trace::UnlockRequested(vid));
+
+        if time_sec > MAX_UNLOCK_TIME_SECS {
+            ringbuf_entry!(Trace::UnlockTimeTooLong(time_sec));
+            return Err(GwMonorailError::TimeIsTooLong);
+        }
+
+        let Some(vlan_index) = VLANS.iter().position(|v| v.vid == vid) else {
+            return Err(GwMonorailError::InvalidVLAN);
+        };
+
+        if VLANS[vlan_index].trusted {
+            return Err(GwMonorailError::AlreadyTrusted);
```

Comment:
Should this be an _error,_ or should it bump the unlock time window forward?

Reply:
I changed the name of this flag to `always_trusted`; it should be an error, because you can't change the trust state of an always-trusted VLAN (which in this case is the Tofino upstream).